pkg/controller/instance/instance_controller.go

```diff
@@ -549,12 +549,12 @@ func getPlanToBeExecuted(i *v1beta1.Instance, ov *v1beta1.OperatorVersion) (*str
 		plan := kudov1beta1.SelectPlan([]string{v1beta1.CleanupPlanName}, ov)
 		if plan != nil {
 			if planStatus := i.PlanStatus(*plan); planStatus != nil {
-				if !planStatus.Status.IsRunning() {
-					if planStatus.Status.IsFinished() {
-						// we already finished the cleanup plan
-						return nil, nil
-					}
+				switch planStatus.Status {
+				case kudov1beta1.ExecutionNeverRun:
```

Comment:
I assume this would be ok, as the the cleanup plan is only ran on the deletion of the instance CR?